pkg/services/hcloud/server/server.go

```diff
@@ -435,6 +435,17 @@ func (s *Service) createServer(ctx context.Context) (*hcloud.Server, error) {
 		return nil, errServerCreateNotPossible
 	}
 
+	if len(sshKeySpecs) == 0 {
+		conditions.MarkFalse(
+			s.scope.HCloudMachine,
+			infrav1.ServerCreateSucceededCondition,
+			"NoSSHKeyFound",
```

Comment:
can you add the reason to the api folder where the other reasons are defined?

Reply:
I saw there is already a reason for that `infrav1.SSHKeyNotFoundReason`